pkg/sysregistriesv2/shortnames_test.go

```diff
@@ -0,0 +1,307 @@
+		SystemRegistriesConfDirPath: "testdata/registries.conf.d",
+		UserShortNameAliasConfPath:  tmp.Name(),
+	}
+
+	configCache = make(map[configWrapper]*V2RegistriesConf)
+	conf, err := TryUpdatingCache(sys)
+	require.NoError(t, err)
+	assert.Len(t, conf.namedAliases, 7)
+	assert.Len(t, conf.Aliases, 0)
+
+	aliases := []struct {
+		name, value string
+	}{
+		{
+			"docker",
+			"docker.io/library/config1", // overriden by config1
+		},
+		{
+			"quay/foo",
+			"quay.io/library/foo",
+		},
+		{
+			"example",
+			"example.com/library/foo",
+		},
+		{
+			"config1",
+			"config1.com/image", // from config1
+		},
+		{
+			"barz",
+			"barz.com/config2", // from config1, overridden by config2
+		},
+		{
+			"config2",
+			"config2.com/image", // from config2
+		},
+		{
+			"added1",
+			"aliases.conf/added1", // from AddShortNameAlias
+		},
+		{
+			"added2",
+			"aliases.conf/added2", // from AddShortNameAlias
+		},
+		{
+			"added3",
+			"aliases.conf/added3", // from config2, overridden by AddShortNameAlias
+		},
+	}
+
+	require.NoError(t, AddShortNameAlias(sys, "added1", "aliases.conf/added1"))
+	require.NoError(t, AddShortNameAlias(sys, "added2", "aliases.conf/added2"))
+	require.NoError(t, AddShortNameAlias(sys, "added3", "aliases.conf/added3"))
+
+	for _, alias := range aliases {
+		resolved, value, err := ResolveShortNameAlias(sys, alias.name)
+		require.NoError(t, err)
+		require.True(t, resolved)
+		assert.NotNil(t, value)
+		assert.Equal(t, alias.value, value.String())
+	}
+
+	resolved, value, err := ResolveShortNameAlias(sys, "i/do/no/exist")
+	require.NoError(t, err)
+	assert.False(t, resolved)
+	assert.Nil(t, value)
+
+	mode, err := GetShortNameMode(sys)
+	require.NoError(t, err)
+	assert.Equal(t, types.ShortNameModePermissive, mode) // from alias.conf, overridden by config2
+
+	// Now remove the aliases from the machine config.
+	require.NoError(t, RemoveShortNameAlias(sys, "added1"))
+	require.NoError(t, RemoveShortNameAlias(sys, "added2"))
+	require.NoError(t, RemoveShortNameAlias(sys, "added3"))
+
+	// Make sure that 1 and 2 are gone.
+	for _, alias := range []string{"added1", "added2"} {
+		resolved, value, err := ResolveShortNameAlias(sys, alias)
+		require.NoError(t, err)
+		assert.False(t, resolved)
+		assert.Nil(t, value)
+	}
+
+	// 3 is still present in config2
+	resolved, value, err = ResolveShortNameAlias(sys, "added3")
+	require.NoError(t, err)
+	assert.True(t, resolved)
+	assert.NotNil(t, value)
+	assert.Equal(t, "xxx.com/image", value.String())
+
+	require.Error(t, RemoveShortNameAlias(sys, "added3")) // we cannot remove it from config2
+}
+
+func TestInvalidAliases(t *testing.T) {
+	sys := &types.SystemContext{
+		SystemRegistriesConfPath:    "testdata/invalid-aliases.conf",
+		SystemRegistriesConfDirPath: "testdata/this-does-not-exist",
+	}
+
+	configCache = make(map[configWrapper]*V2RegistriesConf)
+	_, err := TryUpdatingCache(sys)
+	require.Error(t, err)
+
+	require.Error(t, AddShortNameAlias(sys, "added1", "aliases"))
```

Comment:
Maybe use the temp file for `aliases.conf`, so that we don’t corrupt users’ configuration if any of the validations actually broke.